app/frontend/controllers/tags_controller.js

```diff
@@ -106,21 +106,54 @@ class TagsController {
     new_standard_tag_dlg_options.buttons[i18n.t("general.cancel")] = function() {
       $(this).dialog("close");
     };
-    new_standard_tag_dlg_options.buttons[i18n.t("tags.create-tag")] = function() {
-      tags_controller.saveNewTag(this, "standard");
+    new_standard_tag_dlg_options.buttons[i18n.t("tags.create-tag")] = {
+      id: 'create-tag-button',
+      text: i18n.t("tags.create-tag"),
+      click: function() {
+        tags_controller.saveNewTag(this, "standard");
+      }
     };
   
     $('#new-standard-tag-dialog').dialog(new_standard_tag_dlg_options);
 
     // Handle the enter key in the tag title field and create the tag when it is pressed
-    $('#new-standard-tag-title-input:not(.bound)').addClass('bound').on("keypress", (event) => {
+    $('#new-standard-tag-title-input:not(.bound)').addClass('bound').on("keypress", async (event) => {
       if (event.which == 13) {
+        var tag_title = $('#new-standard-tag-title-input').val();
+        var tag_existing = await this.updateButtonStateBasedOnTagTitleValidation(tag_title, 'create-tag-button');
+
+        if (tag_existing) {
+          return;
+        }
+
         $('#new-standard-tag-dialog').dialog("close");
         tags_controller.saveNewTag(event, "standard");
       }
+    }).on("keyup", async (event) => {
+      var tag_title = $('#new-standard-tag-title-input').val();
+      await this.updateButtonStateBasedOnTagTitleValidation(tag_title, 'create-tag-button');
     });
   }
 
+  async updateButtonStateBasedOnTagTitleValidation(tagTitle, buttonId) {
+    var tag_existing = await this.tagExists(tagTitle);
```

Comment:
Would `tagExists` be fast enough for each `keyup` event?

Reply:
On the desktop - yes!
On Android - I still need to test.

Reply:
Sufficiently fast on Android as well - just tested.